### PR TITLE
Added chart-verifier comes back with failure scenario

### PR DESCRIPTION
**Pull request by tisutisu**
This PR will cover story : https://issues.redhat.com/browse/HELM-243

### CODE REVIEW

File: tests/functional/utils/chart_certification.py [hunk 1]
```diff
@@ -171,6 +171,12 @@ def check_index_yaml(self, base_branch, vendor, chart_name, chart_version, logge
                 f"{chart_name} {chart_version} not added to index")
             return False
 
+        if check_provider_type and self.secrets.vendor_type == 'redhat':
```
Comment (zonggen) [inline tests/functional/utils/chart_certification.py:174]: Can you add a comment on why we are checking this condition? A comment helps when anyone later looks at this condition and wondering why we did this. Also this condition might or might not change so it's beneficial to have a record.

**Comment by zonggen** [comment]
Minor nit otherwise LGTM.

File: tests/functional/utils/chart_certification.py (hunk 1 above)
Comment (tisutisu) [reply tests/functional/utils/chart_certification.py:176]: Agreed. Added comments explaining this check.

Comment (zonggen) [reply tests/functional/utils/chart_certification.py:176]: Thanks! resolving.

**Comment by zonggen** [comment]
Comment was addressed, LGTM.

**Review by zonggen — APPROVED** [review]

**Review by mmulholla — COMMENTED** [review]
The first and 3rd scenario are identical apart from their inputs. Consider reducing to 1.

**Comment by tisutisu** [comment]
> The first and 3rd scenario are identical apart from their inputs. Consider reducing to 1.

Thanks for pointing this out. Merged the first and 3rd scenario into one.

**Review by dperaza4dustbit — APPROVED** [review]
LGTM

**Review by mmulholla — APPROVED** [review]

**Pull request merged**